src/test/java/com/znsio/teswiz/screen/android/vodqa/VodqaScreenAndroid.java

```diff
@@ -82,4 +87,46 @@ public VodqaScreen scrollDownByScreenSize() {
         visually.checkWindow(SCREEN_NAME, "Screen scrolled down");
         return this;
     }
+
+    @Override
+    public VodqaScreen selectScreen(String screenName) {
+        By byScreenNameXpath = AppiumBy.xpath(String.format(screenSelectionXpath, screenName));
+        driver.waitTillElementIsPresent(byScreenNameXpath);
+        driver.findElement(byScreenNameXpath).click();
+        return this;
+    }
+
+    @Override
+    public VodqaScreen swipeLeft() {
+        By bySwipeViewXpath = AppiumBy.xpath(String.format(swipeViewXpath, "1"));
+        if (driver.waitTillElementIsPresent(bySwipeViewXpath).isDisplayed()) {
+            driver.swipeLeft();
+        }
+        return this;
+    }
+
+    @Override
+    public boolean verifySwipe(String elementText) {
+        boolean isSwipeSuccessful = driver.findElement(AppiumBy.xpath(String.format(swipeViewXpath, elementText))).isDisplayed();
+        visually.check(SCREEN_NAME, "Carousel Tile after swipe", Target.region(AppiumBy.xpath(String.format(swipeViewTileXpath, elementText))));
+        return isSwipeSuccessful;
+    }
+
+    @Override
+    public VodqaScreen swipeRight() {
+        By bySwipeViewXpath = AppiumBy.xpath(String.format(swipeViewXpath, "1"));
+        if (driver.waitTillElementIsPresent(bySwipeViewXpath).isDisplayed()) {
```

Comment:
What will happen if this is not displayed? The test does not seem to be in control of its destiny